### PR TITLE
Add a method to TypeSafeEnum that, given a list of TypeSafeEnum instan…

### PR DESCRIPTION
**Pull request by mmartinATX**
…ces, returns a list of their String values

### CODE REVIEW

File: src/classes/TypeSafeEnum.cls [hunk 1]
```diff
@@ -169,4 +169,21 @@ public with sharing abstract class TypeSafeEnum {
...
     }
+
+    /**
+     * Return a list of the non-null values for the type-safe enum instances in the given list
+     *
+     * @param enumInstances the enum instances whose values will be collected
+     *
+     * @return list of the collected values
+     */
+    public static String[] collectStringValues(TypeSafeEnum[] enumInstances) {
```
[the remark this answers is not in the document]
Comment (mmartinATX) [reply src/classes/TypeSafeEnum.cls]: Sure - didn't see that comment in the other pr

File: src/classes/TypeSafeEnum.cls [hunk 2]
```diff
@@ -169,4 +169,23 @@ public with sharing abstract class TypeSafeEnum {
     public static Boolean matchesNone(TypeSafeEnum[] values, String testValue) {
         return !matchesAny(values, testValue);
     }
+
+    /**
+     * Return a list of the non-null values for the <code>TypeSafeEnum</code> instances in the given list
```
Comment (SCWells72) [inline src/classes/TypeSafeEnum.cls]: So basically it returns the non-null values for the non-null entries in the given list, right? Might be good to be more specific about that. Also, please add a period at the end of the sentence.

File: src/classes/TypeSafeEnum.cls [hunk 3]
```diff
@@ -169,4 +169,23 @@ public with sharing abstract class TypeSafeEnum {
...
+     *
+     * @param enumInstances the enum instances whose values will be collected
+     *
+     * @return list of the enum's <code>String</code> values
+     */
+    public static String[] toStrings(TypeSafeEnum[] enumInstances) {
+        String[] valueStrings = new String[]{};
```
Comment (SCWells72) [inline src/classes/TypeSafeEnum.cls]: Since this is being mutated, can you make it `List<String> values = new List<String>();`?

Comment (mmartinATX) [reply src/classes/TypeSafeEnum.cls]: Sure

**Pull request merged**